### DISCUSSION TITLE
Copy-PnPFile stopped working

### DISCUSSION THREAD

**Discussion by nazihhaddad** [pnp/powershell#125] [Q&A]
Hello,

The following command stopped working for me since two weeks, it was working perfectly with no issues.

$SiteURL = "https://tenant.sharepoint.com/sites/SourceSite"
$SourceFolderURL = "/sites/SourceSite/Public"
$TargetFolderURL = "/sites/TargetSite/Public"
Connect-PnPOnline -Url $SiteURL -Credential $MyCreds
Copy-PnPFile -SourceUrl $SourceFolderURL -TargetUrl $TargetFolderURL -SkipSourceFolderName -Force -OverwriteIfAlreadyExists

Copy-PnPFile : Unknown Error
At line:1 char:1
+ Copy-PnPFile -SourceUrl $SourceFolderURL -TargetUrl $TargetFolderURL  ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Copy-PnPFile], ServerException
    + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.Files.CopyFile

Could anyone please assist?

Thank you,

**Comment by erwinvanhunen** [comment]
Which version of PnP PowerShell are you using?

**Comment by stuartcporter** [comment]
hi
it's no longer working for me - and i just installed the latest build!
should this not work?
 PS C:\Users\sporter> Copy-PnPFile -SourceUrl "/Shared Documents" -TargetUrl "/sites/site1/Shared Documents" -OverwriteIfAlreadyExists
Copy-PnPFile : {"odata.error":{"code":"-2147467261, System.ArgumentNullException","message":{"lang":"en-US","value":"Value cannot be null.\r\nParameter name: Null value  

 Manifest   1.3.0      PnP.PowerShell 

any ideas?

**Reply by stuartcporter** [reply]
seems to be some odd behaviour going on  

this works going from one site to another site 
Copy-PnPFile -SourceUrl "/sites/site2/Shared Documents/test2" -TargetUrl "/sites/site1/Shared Documents" -
OverwriteIfAlreadyExists

however this copying between two libraries in the same site ... 
Copy-PnPFile -SourceUrl "/sites/site2/Shared Documents/test2" -TargetUrl "/sites/site2/copydest/" -OverwriteIfAlreadyExists
fails with a 
Copy-PnPFile : Access denied. You do not have permission to perform this action or access this resource. 

even thou account has full control ?

**Comment by veronicageek** [comment]
@stuartcporter - The following works for copying from one library to another within the _same_ site:

```powershell
$sourceUrl = "/sites/sitepnp1/Shared Documents/TestDoc.docx"
$targetUrl = "/sites/sitepnp1/TargetLib/TestDoc.docx"

Copy-PnPFile -SourceUrl $sourceUrl -TargetUrl $targetUrl -OverwriteIfAlreadyExists -Force
```
OR this:

```powershell
$sourceUrl = "Shared Documents/TestDoc.docx"
$targetUrl = "/sites/sitepnp1/TargetLib/TestDoc.docx"

Copy-PnPFile -SourceUrl $sourceUrl -TargetUrl $targetUrl -OverwriteIfAlreadyExists -Force
```
